### ISSUE TITLE
Customize / extend Prompt

### ISSUE THREAD

**Issue by giuliano-abinbev** [warpdotdev/Warp#3957]
### Describe the solution you'd like?

```Text
I would be nice to be able to extend Warp Prompt, I know that the user have 100% customize it via shell prompt but being able to customize Warp Prompt would be amazing
```

### How important is this feature to you?

3

### Warp Internal (ignore) - linear-label:770f6576-d6c0-4e4f-a259-fc64b5156087

None

**Comment by dannyneira** [comment]
Hi @giuliano-abinbev Thanks for submitting. There are several customizations you can make to the Warp Prompt. Please see our [docs for more](https://docs.warp.dev/appearance/prompt).

Do you want to be able to add your custom context chip to the Warp prompt? If there is something specifically missing please let us know so we can have a better idea of what this request is meant to address. 

To anyone else interested in this feature, please add a :+1: to the original post at the top to signal that you want this feature, and subscribe if you'd like to be notified.

**Comment by giuliano-abinbev** [comment]
@dannyneira correct, and it would be nice to be able to extend Warp Prompt, I like how you folks simplified it, I just drag and drop and done, plus I don't end up using some tool that ends up adding load time to my terminal.

Currently using [bit.dev](https://bit.dev/) for one of my projects and I'm exploring a solution to display the tool's info instead of `git`

**Comment by mrmelon54** [comment]
Would be amazing if I could add conditional parts to the prompt.

e.g. Should start and end time only if the command takes longer than X seconds to run.

I don't want to see times if I'm running quick commands but for long running tasks I want to know when it started and ended.

Add custom segments for warp prompt using shell scripts. Basically taking the scripts that generate a custom PS1 and putting them as segments inside warp.

Maybe allow displaying warp segments as well as PS1

**Comment by igorsantos07** [comment]
The idea to have custom chips based on script output is just awesome for easy extensibility!

I came down here looking for a way to configure some chips - e.g., I _know_ the yellow chip is the git branch, there's no need for the `git:(...)` wrapping the branch name - that could be made shorter. I guess the same could be true for the SVN chip; time chip could have a 12/24 option and show seconds option, instead of having two separate chips, etc. Same for date format.

**Comment by adexot** [comment]
> Hi [@giuliano-abinbev](https://github.com/giuliano-abinbev) Thanks for submitting. There are several customizations you can make to the Warp Prompt. Please see our [docs for more](https://docs.warp.dev/appearance/prompt).
> 
> Do you want to be able to add your custom context chip to the Warp prompt? If there is something specifically missing please let us know so we can have a better idea of what this request is meant to address.
> 
> To anyone else interested in this feature, please add a :+1: to the original post at the top to signal that you want this feature, and subscribe if you'd like to be notified.

Hi @dannyneira 👋🏾 

> If there is something specifically missing please let us know so we can have a better idea of what this request is meant to address.

Can you please add a chip for the current git user name
